src/transformers/models/bert/modeling_flax_bert.py

```diff
@@ -890,63 +909,115 @@ def __call__(
         if head_mask is None:
             head_mask = jnp.ones((self.config.num_hidden_layers, self.config.num_attention_heads))
 
-        # Handle any PRNG if needed
-        rngs = {}
-        if dropout_rng is not None:
-            rngs["dropout"] = dropout_rng
+        if self.scope is None:
```

Comment:
So `self.scope` becomes our overarching way of telling whether the params should/shouldn't be passed? Is this better than `self._do_init`?

Reply:
It seems there are 3 ways this class would operate right now:
1. `_do_init=True`, the stafeful way.
2. `_do_init=False` but passing `model(..., params=params)`, the current "functional" way.
3. `_do_init=False` but using `mode.apply`, this results in `self.scope != None`.

Reply:
Right I see, thanks for clarifying!